### ISSUE TITLE
Behaviour of minimum_supported_version and replacement functions

### ISSUE THREAD

**Issue by johnbillion** [WordPress-Coding-Standards/WordPress-Coding-Standards#1113]
A plugin of mine still supports WordPress version 3.7, therefore it uses `parse_url()` instead of `wp_parse_url()` which was introduced in 4.4.

In my plugin's `phpcs.xml.dist` file I have the following:

```xml
<rule ref="WordPress.WP.DeprecatedFunctions">
	<properties>
		<property name="minimum_supported_version" value="3.7" />
	</properties>
</rule>
```

Despite this, a warning is still reported about the use of `parse_url()`.

Is there a means for WPCS to respect the `minimum_supported_version` property when recommending the use of replacement functions, in addition to its behaviour for deprecated functions?

**Comment by jrfnl** [comment]
@johnbillion Not at this time, but I like the idea. Tagging this as a feature request.

**Comment by jrfnl** [comment]
@johnbillion I've had a more detailed look at this and your issue confuses me.

In your description, you talk about `parse_url()` and the `DeprecatedFunctions` sniff.

The `DeprecatedFunctions` sniff does not sniff for `parse_url()`, it only sniffs for WP native deprecated functions.

The `AlternativeFunctions` sniff, however, *does* sniff for `parse_url()` and recommends using `wp_parse_url()` instead.
The `AlternativeFunctions` sniff takes the `minimum_supported_version` into account when this has been set [for all sniffs](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/wiki/Customizable-sniff-properties#setting-minimum-supported-wp-version-for-all-sniffs-in-one-go-wpcs-0140) and when set for [the `AlternativeFunctions` sniff via the property](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/wiki/Customizable-sniff-properties#minimum-wp-version-to-check-for-usage-of-deprecated-functions-classes-and-function-parameters).

This was build in via #1285 and #1369, both released in WPCS 1.0.0.

So, if your issue was with the `AlternativeFunctions` sniff, I believe it has been fixed in the mean time and this issue should be closed.

If your issue is with the `DeprecatedFunctions` sniff, we need to look further.

That sniff throws an `error` when a function is used which was deprecated in a WP version _before_ the `minimum_supported_version`.
It throws a `warning` otherwise.
It will also recommend alternatives (if known).

What the sniff does _not_ do, is check if the alternative is available in your `minimum_supported_version` and stay silent in the case it isn't.

To add that functionality, a new `alt_since` key would need to be added to the array  for any deprecated function which has an alternative, keeping track of when the alternative was added to WP.
The same would need to be done for the `DeprecatedClasses` and the `DeprecatedParameterValues` sniffs.

All the same, I would expect that situation to be quite rare as most of the time, the deprecation and the adding of the alternative are done in the same WP release (or the deprecation is done in a later release), so I expect there rarely to be a problem with this for those sniffs.

What do you think ?

**Comment by johnbillion** [comment]
My original question was:

> Is there a means for WPCS to respect the minimum_supported_version property when recommending the use of replacement functions, in addition to its behaviour for deprecated functions?

Which has since been implemented, so this can be closed.
